fennel/datasets/datasets.py

```diff
@@ -1924,6 +1924,11 @@ def is_subset(subset: List[str], superset: List[str]) -> bool:
                 )
             # Check the schemas of the keys
             for key in obj.on:
+                if fennel_is_optional(left_schema.get_type(key)):
+                    raise TypeError(
+                        f"Optional typed fields are not allowed as Key field in left schema got `{key}` "
```

Comment:
I dont think Key field is the right terminology since in a join the fields on the LHS can be non-key fields too. 

Something along the lines of - , Fields used in a join operation must not be optional, found x of type Optional[y] in Pipeline xyz ?

Reply:
makes sense, refracted the error statement

Reply:
Also increased the version.